spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_stdp.py

```diff
@@ -508,10 +508,12 @@ def get_max_synapses(self, n_words):
         n_connections = (n_words_space * BYTES_PER_WORD) // (
             bytes_per_pp + bytes_per_fp)
 
-        # Reduce until correct
-        while (self.get_n_words_for_plastic_connections(n_connections) >
-               n_words):
-            n_connections -= 1
+        # Reduce until correct, if padding (i.e. structural plasticity)
+        # is not present (otherwise this can underflow)
+        if self.__pad_to_length is None:
+            while (self.get_n_words_for_plastic_connections(n_connections) >
+                   n_words):
+                n_connections -= 1
```

Comment:
Not quite right - this will give the wrong answer unfortunately, as this is supposed to return the maximum number of connections supported

Reply:
Yes, but the error is already somewhat nonsensical in this case anyway as it can plausibly say there are too many synapses (0) (i.e. you've started with a FixedProbabilityConnector(p=0.0)) than can be supported.

I'm still a bit confused here, to be honest.  Are we supposed to be supporting s_max > 256?  If not, then why not just throw an exception immediately... ?

Reply:
> I'm still a bit confused here, to be honest. Are we supposed to be supporting s_max > 256? If not, then why not just throw an exception immediately... ?

s_max refers to fan in. Meaning, each neuron can have up to s_max (let's say 500) incoming connections. This does not necessarily break the assumption that the length of a synaptic row is greater than `256`. For example, this neuron with a maximum fan in of 500 could be receiving a synapse from 500 pre-synaptic neurons (1 from each), thus the length of individual rows would be 1. 

I originally used s_max as the estimate for pre-synaptic population fan-out, and thus length of synaptic row. This should be done better.

Reply:
OK, thanks - that makes more sense now.

I think the most recent commit fixes the hanging issue; so we should probably merge that and then maybe consider how to do this better in the future.